pkg/internal/discover/typer.go

```diff
@@ -146,12 +152,19 @@ func (t *typer) asInstrumentable(execElf *exec.FileInfo) ebpf.Instrumentable {
 
 	detectedType := exec.FindProcLanguage(execElf.Pid, execElf.ELF, execElf.CmdExePath)
 
+	if detectedType == svc.InstrumentableGolang && err == nil {
```

Comment:
This particular bit relates to this:
>There is a small corner case in which we detect no go offsets in a target executable, but if that executable happens to have a ".gosyms" section, we will end up classifying it as a Go executable, causing issues. We attach an error in this case to have the attacher treat it as a generic executable.

It's a very corner case, not present in the reported issue but I discovered it while analysing the code. If offsets are null (and thus 'InstrumentationError` is also null), but for some esoteric reason the target executable has a `.gosymtab`, this will cause `detectedType` to  be `InstrumentableGolang`, it will trigger [this codepath](https://github.com/grafana/beyla/blob/119226432feff14ed60bd404f615d0ee84e3c671/pkg/internal/discover/attacher.go#L153) and crash with a null offsets pointer.

Reply:
Ah I see now, okay makes sense.